c10/core/TensorImpl.cpp

```diff
@@ -95,6 +95,10 @@ TensorImpl::TensorImpl(Storage&& storage, DispatchKeySet key_set, const caffe2::
     key_set_ = key_set | getAutogradRelatedKeySetFromBackend(k);
   }
 
+  if (is_conj()) {
+    TORCH_INTERNAL_ASSERT(isComplexType(typeMetaToScalarType(dtype())));
```

Comment:
@ezyang why do we need this?

Reply:
Internal invariant. Conjugate bit is not valid on non-complex tensors.